src/Measurement/Energy.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Sundata\Utilities\Measurement;
+
+class Energy implements Measurement
+{
+    private float $wattHour;
+    public int $precision = 2;
+
+    private function __construct(
+        $wattHour
+    ) {
+        $this->wattHour = $wattHour;
+    }
+
+    public static function fromKwh(float $kwh): Energy
+    {
+        return new Energy($kwh * 1000);
+    }
+
+    public function asWh(): float
+    {
+        return $this->wattHour;
+    }
+
+    public function asKwh(): float
+    {
+        return $this->wattHour / 1E3;
+    }
+
+    public function asMwh(): float
+    {
+        return $this->wattHour / 1E6;
+    }
+
+    public function asTwh(): float
+    {
+        return $this->wattHour / 1E6;
+    }
+
+    public function format(): FormattedMeasurement
+    {
+        if ($this->wattHour >= 1E13) {
```

Comment:
er zitten fouten in deze functie. misschien een array gebruiken met de 'settings'. dat lijkt me minder fout gevoelig

Reply:
Ik heb de test herschreven en waardoor alles leesbaarder is.